src/Cube/Heuristics/StateOne.hs

```diff
@@ -1,9 +1,112 @@
-module Cube.Heuristics.StateOne
-( binaryFold
-) where
+module Cube.Heuristics.StateOne where
+
+import Cube.Edges
+import Cube.Moves ((-:))
+import Cube.Moves.Edges
+import Data.Char
+import qualified Data.HashMap.Strict as M
+import qualified Data.Vector.Unboxed as V
+
+{-
+ - A list of all 18 possible moves under the half-turn metric (HTM)
+ -}
+moves :: [Edge -> Edge]
+moves = [
+      front
+    , back
+    , left
+    , right
+    , up
+    , down
+    , front . front
+    , back . back
+    , left . left
+    , right . right
+    , up . up
+    , down . down
+    , front . front . front
+    , back . back . back
+    , left . left . left
+    , right . right . right
+    , up . up . up
+    , down . down . down
+    ]
+
+{-
+ - Used to unfold a hashmap key to a list of integers.
+ - It just converts the numberic character at each index to its integer equivalent
+ -}
+unfoldToInt :: String -> Maybe (Int, String)
+unfoldToInt [] = Nothing
+unfoldToInt (x:xs) = Just (digitToInt x, xs)
+
+{-
+ - Used to fold a list of integers to a string.
+ - It just prepends the character value of the integer to the string.
+ - NOTE: Unfortunately this produces a string representation of the vector but in reverse order. This was specifically
+ - chosen because it is used extensively in the solve algorithm so reversing the vector is unnecessary overhead.
+ -}
+foldString :: String -> Int -> String
+foldString carry value = intToDigit value : carry
+
+{-
+ - Static lookup list. This is to be generated from a script ... TODO
```

Comment:
Fix TODO

How to have the map serialised so its only generated once at build time or something similar